docs/developing/languages/ruby/installation.md

```diff
@@ -16,52 +12,131 @@ for several reasons:
 
-- Install Ruby manually
+It's also possible to use Docker, instructions for which are at the bottom of this document.
 
-### Allows switching between multiple versions of Ruby
+### Setup Instructions
 
-- Install and configure a Ruby manager, then install Ruby
+You'll need to know if the project you are running requires a specific version of Ruby. The convention is for this to be defined in a project's `Gemfile`. Look for a line that looks like this:
 
-- Use a vetted automated script that sets everything up for you
+```ruby
+ruby "~> 2.7.2"
+```
 
-## The Trussel Way
+In the above example, any version of Ruby in the range `2.7.2` to `2.7.99999` will suffice.
 
-For long-term day-to-day development, we recommend installing and using Ruby via
-a Ruby manager. Inevitably, when working with Ruby projects, you will need to
-install multiple different versions at the same time and switch between them.
+#### Install Prerequisites
 
-Depending on your preferences and needs, the two recommended ways to install and
-use Ruby on a Mac are via local installation of a Ruby manager, or via Docker.
+- Install [Homebrew](https://brew.sh/)
+- Install Apple Command Line Tools using `xcode-select --install`
 
-Reasons to use Docker:
+#### Install `asdf`
 
-- You don't use Ruby regularly, and only need it occasionally, such as to
-  review candidate work samples
-- You don't want to install anything locally
+[asdf](https://asdf-vm.com) is a program that manages multiple language versions. It uses language-specific plugins to define the specifics for each language it supports. We'll be using [the officially supported plugin for Ruby](https://github.com/asdf-vm/asdf-ruby) in this guide.
+
+Install asdf using homebrew:
+
+```console
+$ brew install asdf
+```
+
+Then, configure your shell to load asdf. The following instructions are for ZSH (the default shell in macOS); for other shells or OSes, find the relevant section in [the asdf installation documentation](https://asdf-vm.com/guide/getting-started.html#_3-install-asdf).
+
+```console
+$ echo -e "\n. $(brew --prefix asdf)/libexec/asdf.sh" >> ${ZDOTDIR:-~}/.zshrc
+```
+
+Close your shell and open a new one. Type `asdf` and you should see something like the following:
+
+```console
+$ asdf
+version: v0.10.2
+
+MANAGE PLUGINS
+asdf plugin add <name> [<git-url>]      Add a plugin from the plugin repo OR,
+                                        add a Git repo as a plugin by
+                                        specifying the name and repo url
+
+...
+```
+
+If that is working, continue on to the next step.
+
+#### Install asdf-ruby
+
+To add support for Ruby to asdf, add the plugin:
+
+```console
+$ asdf plugin add ruby https://github.com/asdf-vm/asdf-ruby.git
+```
+
+Then ensure that you have a build environment setup to install Ruby:
+
+```console
+$ brew install openssl@1.1 readline
+$ export RUBY_CONFIGURE_OPTS="--with-openssl-dir=$(brew --prefix openssl@1.1)"
```

Comment:
I didn't have to do this?

Reply:
OK. I confirmed that Ruby will bundle its own openssl here, at least for 2.7.6. Do we *want* to use home-brew's for a specific reason?

Reply:
It seems that installing `asdf` via home-brew will pull in openssl and readline anyway, so the first line is redundant:

```console
$ brew info asdf
asdf: stable 0.10.2 (bottled), HEAD
...
==> Dependencies
Required: autoconf ✔, automake ✔, coreutils ✔, libtool ✔, libyaml ✔, openssl@1.1 ✔, readline ✔, unixodbc ✔
```

As far as which openssl to use, [the current release of `openssl-1.1.1q` doesn't build on macOS](https://github.com/openssl/openssl/issues/18720) (I'm on Intel and also seeing this). I end up getting this when I run `asdf install ruby 3.1.2`:

```console
$ asdf install ruby 3.1.2
Downloading openssl-1.1.1q.tar.gz...
-> https://dqw8nmjcqpjn7.cloudfront.net/d7939ce614029cdff0b6c20f0e2e5703158a489a72b2507b8bd51bf8c8fd10ca
Installing openssl-1.1.1q...

BUILD FAILED (macOS 12.4 using ruby-build 20220710)

Inspect or clean up the working tree at /var/folders/7l/st6gd61x5bx1t0wsh3m5prb80000gn/T/ruby-build.20220711103203.48433.CyXqzX
Results logged to /var/folders/7l/st6gd61x5bx1t0wsh3m5prb80000gn/T/ruby-build.20220711103203.48433.log

Last 10 log lines:
clang  -Iinclude -arch x86_64 -O3 -Wall -D_REENTRANT -DZLIB -DZLIB_SHARED -DNDEBUG -I/Users/jimb/.asdf/installs/ruby/3.1.2/include  -MMD -MF test/x509_time_test.d.tmp -MT test/x509_time_test.o -c -o test/x509_time_test.o test/x509_time_test.c
clang  -Iinclude -arch x86_64 -O3 -Wall -D_REENTRANT -DZLIB -DZLIB_SHARED -DNDEBUG -I/Users/jimb/.asdf/installs/ruby/3.1.2/include  -MMD -MF test/x509aux.d.tmp -MT test/x509aux.o -c -o test/x509aux.o test/x509aux.c
test/v3ext.c:201:24: error: implicitly declaring library function 'memcmp' with type 'int (const void *, const void *, unsigned long)' [-Werror,-Wimplicit-function-declaration]
        if (!TEST_true(memcmp(ip1->data, ip2->data, ip1->length) <= 0))
                       ^
test/v3ext.c:201:24: note: include the header <string.h> or explicitly provide a declaration for 'memcmp'
1 error generated.
make[1]: *** [test/v3ext.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make: *** [all] Error 2
```

[There is a fix in for openssl already](https://github.com/openssl/openssl/pull/18719), but there isn't a new release out yet. `ruby-build` installs openssl unless `--with-openssl-dir` is set within `RUBY_CONFIGURE_OPTS`, so using the homebrew version works around this and installs whatever version is in homebrew (which presumedly at least compiles). I think the homebrew version is less likely to fail to install than one that's compiled from source, so I'd opt to keep relying on openssl from brew.

Reply:
Works for me. Let's make sure there's something in the doc that explains why so that future impatient folks don't skip this step.